reactor-core/src/main/java/reactor/core/Exceptions.java

```diff
@@ -422,13 +490,16 @@ public static <T> Throwable terminate(AtomicReferenceFieldUpdater<T, Throwable>
 	 * @param t the exception to evaluate
 	 */
 	public static void throwIfFatal(@Nullable Throwable t) {
-		if (t instanceof BubblingException) {
-			throw (BubblingException) t;
-		}
-		if (t instanceof ErrorCallbackNotImplemented) {
-			throw (ErrorCallbackNotImplemented) t;
+		if (t == null) {
+			return;
 		}
+		//we give throwIfJvmFatal an opportunity to detect, log and throw first
 		throwIfJvmFatal(t);
```

Comment:
With this logic, the `isJvmFatal` is called twice for non-jvm-fatal exceptions - first by `throwIfJvmFatal` and then by `isFatal` in the next line. Perhaps having two methods:
`isJvmFatal` and `isNonJvmFatal` would allow the implementations of `throwIfFatal` and `throwIfJvmFatal` to be a bit cleaner?

Reply:
I added a non-public isFatalButNotJvmFatal so that the `throwIfFatal` code is cleaner/clearer while still having each exception type tested in a single place 👍